huaweicloud/resource_huaweicloud_fgs_function_v2.go

```diff
@@ -227,10 +358,87 @@ func resourceFgsFunctionV2Read(d *schema.ResourceData, meta interface{}) error {
 	} else {
 		d.Set("xrole", f.Xrole)
 	}
+	d.Set("app_xrole", f.AppXrole)
+	d.Set("depend_list", f.DependList)
+	d.Set("initializer_handler", f.InitializerHandler)
+	d.Set("initializer_timeout", f.InitializerTimeout)
+	d.Set("enterprise_project_id", f.EnterpriseProjectID)
+
+	// set func_vpc
+	if f.FuncVpc != (function.FuncVpc{}) {
+		funcVpc := []map[string]string{
+			{
+				"vpc_id":      f.FuncVpc.VpcId,
+				"vpc_name":    f.FuncVpc.VpcName,
+				"subnet_id":   f.FuncVpc.SubnetId,
+				"subnet_name": f.FuncVpc.SubnetName,
+				"cidr":        f.FuncVpc.Cidr,
+				"gateway":     f.FuncVpc.Gateway,
+			},
+		}
+		d.Set("func_vpc", funcVpc)
+	}
+
+	// set mount_config
+	if f.MountConfig.MountUser != (function.MountUser{}) {
+		mountUser := []map[string]int{
+			{
+				"user_id":       f.MountConfig.MountUser.UserId,
+				"user_group_id": f.MountConfig.MountUser.UserGroupId,
+			},
+		}
+		funcMounts := make([]map[string]string, 0, len(f.MountConfig.FuncMounts))
+		for _, v := range f.MountConfig.FuncMounts {
+			funcMount := map[string]string{
+				"id":               v.Id,
+				"mount_type":       v.MountType,
+				"mount_resource":   v.MountResource,
+				"mount_share_path": v.MountSharePath,
+				"local_mount_path": v.LocalMountPath,
+				"status":           v.Status,
+			}
+			funcMounts = append(funcMounts, funcMount)
+		}
+		funcMountConfig := []map[string]interface{}{
+			{
+				"mount_user":  mountUser,
+				"func_mounts": funcMounts,
+			},
+		}
+		d.Set("mount_config", funcMountConfig)
+	}
 
 	return nil
 }
 
+func resourceFgsFunctionV2Update(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*config.Config)
+	fgsClient, err := config.FgsV2Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("Error creating HuaweiCloud FGS V2 client: %s", err)
+	}
+
+	urn := d.Id()
+	if strings.HasSuffix(urn, ":latest") {
+		urn = urn[0 : len(urn)-7]
+	}
+
+	//lintignore:R019
+	if d.HasChanges("code_type", "code_url", "code_filename", "depend_list", "func_code") {
+		err := resourceFgsFunctionV2CodeUpdate(fgsClient, urn, d)
+		if err != nil {
+			return err
+		}
+	} else {
```

Comment:
if both `code_type` and metadata have changed, only `code_type` will be updated

Reply:
fixed